src/test/java/io/debezium/connector/yugabytedb/consistent/MessageTest.java

```diff
@@ -34,4 +60,153 @@ void comparatorTest() {
         assertEquals(1, commit.compareTo(insert));
     }
 
+    @Test
+    public void checkForBeginMessages() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.BEGIN)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.isBegin(m));
+    }
+
+    @Test
+    public void checkForCommitMessages() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.COMMIT)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.isCommit(m));
+    }
+
+    @Test
+    public void checkNeitherBeginNorCommit() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.INSERT)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m1 = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+        Message m2 = new Message.Builder().setRecord(record).setTabletId("anotherDummyTablet")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.notBeginCommit(m1, m2));
+    }
+
+    @Test
```

Comment:
One disadvantage of this approach is that it is sometimes unclear which of the combinations failed. A better structure is:

```
@DisplayName("test different combinations for compareTo")
@ParameterizedTest(name: "{index} => this: ({0}, {1}, {2}), op:({3},{4},{5})"
@MethodSource("messageProvider")
void compareToTest(CdcService.RowMessage.Op op1, 
                                   CdcService.RowMessage.Op op2, 
                                   long commitTime1, 
                                   long commitTime2, 
                                   long recordTime1, 
                                   long recordTime2, 
                                   long expectedResult)
                                   {
// Build messages
    assertEquals(m1.compareTo(m2), this.expectedResult)
}

private static Stream<Arguments> messageProvider() {
        return Stream.of(
                Arguments.of(Op.BEGIN, Op.INSERT, lowCommitTime, highCommitTime, 0, lowRecordTime, -1),
                Arguments.of(Op.BEGIN, Op.INSERT, highCommitTime, lowCommitTime, 0, lowRecordTime, 1),
                ....
        );
    }
```

Here is a canonical example:

```
@ParameterizedTest
@MethodSource("stringIntAndListProvider")
void testWithMultiArgMethodSource(String str, int num, List<String> list) {
    assertEquals(5, str.length());
    assertTrue(num >=1 && num <=2);
    assertEquals(2, list.size());
}

static Stream<Arguments> stringIntAndListProvider() {
    return Stream.of(
        arguments("apple", 1, Arrays.asList("a", "b")),
        arguments("lemon", 2, Arrays.asList("x", "y"))
    );
}
```
Reference: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource